ibis/expr/analysis.py

```diff
@@ -167,15 +170,9 @@ def fn(node):
 
 def substitute_unbound(node):
     """Rewrite `node` by replacing table expressions with an equivalent unbound table."""
-    assert isinstance(node, ops.Node), type(node)
-
-    def fn(node, _, **kwargs):
-        if isinstance(node, ops.DatabaseTable):
-            return ops.UnboundTable(name=node.name, schema=node.schema)
-        else:
-            return node.__class__(**kwargs)
-
-    return node.map(fn)[node]
+    return node.replace(
+        p.DatabaseTable(name=x, schema=y) >> c.UnboundTable(name=x, schema=y)
```

Comment:
Can we instantiate the pattern variables in this function's scope instead of globally, and also call them `name` and `schema`?

Reply:
We could just define the two variables in the function or we could use capture and variable object directly in the pattern too. Matter of preference mostly, but we can revisit it later.